commit 3b33a0b426b8915f302247dcd40c64fb1a6dcdd2
Author: Emil Velikov <emil.l.velikov@gmail.com>
Date:   Sun Nov 4 18:06:18 2012 +0000

    kms: Let the server know if we got more than 1 EDID block
    
    Verbatim copy from xf86-video-radeon commit 809cacb
    
    Bugzilla: https://bugs.freedesktop.org/show_bug.cgi?id=32341
    Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>
    Signed-off-by: Ben Skeggs <bskeggs@redhat.com>

diff --git a/src/drmmode_display.c b/src/drmmode_display.c
index 803785d..8ec69f5 100644
--- a/src/drmmode_display.c
+++ b/src/drmmode_display.c
@@ -640,9 +640,12 @@ drmmode_output_get_modes(xf86OutputPtr output)
 		drmModeFreeProperty(props);
 	}
 
-	if (drmmode_output->edid_blob)
+	if (drmmode_output->edid_blob) {
 		ddc_mon = xf86InterpretEDID(output->scrn->scrnIndex,
 					    drmmode_output->edid_blob->data);
+		if (ddc_mon && drmmode_output->edid_blob->length > 128)
+			ddc_mon->flags |= MONITOR_EDID_COMPLETE_RAWDATA;
+	}
 	xf86OutputSetEDID(output, ddc_mon);
 
 	/* modes should already be available */
